pbt-gen/Cargo.toml

```
@@ -0,0 +1,16 @@
+[package]
+name = "pbt-gen"
```

We typically prefix all crates' names within this repository with `tendermint-`, like:

```suggestion
name = "tendermint-pbt-gen"
```

---

I thought we might omit the long prefix since it's just for internal use, but happy to follow the convention if that matters here.

It will make it awkward to call functions using the current convention in the examples (where the short crate name is used as an accessor, like `pbt_gen::time::arb_datetime()`. Would you recommend I rename the `time` module to something less general, or do you think we'll end up just importing the particular functions directly?

---

I think both names are fine as long as it does not depend on anything from `tendermint`, in case I would argue for the `tendermint-` prefix.

Regarding the issue of short crate name for easy invocations, you can rename a crate when you specify it as a dependency in `Cargo.toml`. The following will add the `tendermint-pbt-gen` crate as a dependency and let you use it as `pbt_gen` in the code.

```toml
[dependencies]
pbt-gen = { package = "tendermint-pbt-gen", version = "0.1" }
```

See https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#renaming-dependencies-in-cargotoml

---

I think this cannot depend on anything from `tendermint`, because `tendermint` depends on it (as a dev dependency), and, iiuc, crate dependencies cannot be circular?

Nice to learn about dep renaming! Thanks :) I think it may be clearer to just rename the crate on `use` tho. I'll try that and see how that sits with folks.